pkg/runtime/ast/ast.go

```diff
@@ -1757,3 +1759,44 @@ func (cc *convCtx) convKill(stmt *ast.KillStmt) Statement {
 		ConnectionID: stmt.ConnectionID,
 	}
 }
+
+// Convert mysql optimizer hints
+// Include https://dev.mysql.com/doc/refman/8.0/en/optimizer-hints.html#optimizer-hints-index-level
+func (cc *convCtx) convTableHint(stmt []*ast.TableOptimizerHint) *HintNode {
+	// ignore hints filter
+	var isIgnore = func(hintName string) bool {
+		ignoreMap := map[string]string{
+			"TIDB_HJ": "tidb hj hints",
+		}
+		if _, ok := ignoreMap[hintName]; ok {
+			return true
+		}
+		return false
+	}
+
+	hints := make([]HintItem, 0, len(stmt))
+	for _, hintStmt := range stmt {
+		sb := strings.Builder{}
+		// restore by parser
+		err := hintStmt.Restore(format.NewRestoreCtx(format.DefaultRestoreFlags, &sb))
+		if err != nil {
+			continue
+		}
+		if isIgnore(hintStmt.HintName.String()) {
+			continue
+		}
+		hintItem := HintItem{
+			TP:       MysqlHint,
+			HintExpr: sb.String(),
+		}
+		hints = append(hints, hintItem)
+	}
+
+	if len(hints) == 0 {
+		return nil
```

Comment:
nil or empty struct?

Reply:
nil will be more conviniewnt to check i think